shared_model/interfaces/commands/create_role.hpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_SHARED_MODEL_CREATE_ROLE_HPP
+#define IROHA_SHARED_MODEL_CREATE_ROLE_HPP
+
+#include <set>
+#include <numeric>
+#include "interfaces/common_objects/types.hpp"
+#include "interfaces/hashable.hpp"
+#include "model/commands/create_role.hpp"
+
+namespace shared_model {
+  namespace interface {
+    /**
+     * Create new role in Iroha
+     */
+    class CreateRole : public Hashable<CreateRole, iroha::model::CreateRole> {
+     public:
+      /**
+       * @return Id of the domain to create
+       */
+      virtual const types::RoleIdType &roleName() const = 0;
+      /// Set of Permissions
```

Comment:
Maybe here we may describe, why we use `set`, but not `unordered_set` here? This is just a suggestion, you may ignore this issue.